Content.Server/Database/PostgresNotificationManager.cs

```diff
@@ -0,0 +1,162 @@
+        _connection.Notification += OnNotification;
+
+        var cancellationToken = _tokenSource.Token;
+        Task.Run(() => NotificationListener(cancellationToken), cancellationToken);
+    }
+
+    /// <summary>
+    /// Listens to the notification channel with basic error handling and reopens the connection if it got closed
+    /// </summary>
+    private async Task NotificationListener(CancellationToken cancellationToken)
+    {
+        if (_connection == null)
+            return;
+
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            if (_connection.State == ConnectionState.Broken)
+                await _connection.CloseAsync();
+
+            if (_connection.State == ConnectionState.Closed)
+            {
+                await _connection.OpenAsync(cancellationToken);
+            }
+
+#if EXCEPTION_TOLERANCE
+            try
+            {
+#endif
+                await using var cmd = new NpgsqlCommand($"LISTEN {BanNotificationChannel}", _connection);
+                await cmd.ExecuteNonQueryAsync(cancellationToken);
+                await _connection.WaitAsync(cancellationToken);
+#if EXCEPTION_TOLERANCE
+            }
+            catch (NpgsqlException e)
+            {
+                _runtimeLog.LogException(e, $"{nameof(PostgresNotificationManager)}.${nameof(NotificationListener)}");
+            }
+#endif
+        }
+    }
+
+    private void OnNotification(object _, NpgsqlNotificationEventArgs notification)
+    {
+        _logger?.Info($"Received postgres notification. Channel: {notification.Channel} Payload: {notification.Payload}");
+
+        // Only one channel so just an if
+        if (notification.Channel != BanNotificationChannel)
+            return;
+
+        var notificationData = JsonSerializer.Deserialize<BanNotificationData>(notification.Payload);
+        if (notificationData == null)
+            return;
+
+        // ReSharper disable once AsyncVoidLambda
+        _taskManager.RunOnMainThread(async () => await OnBanNotification(notificationData));
+    }
+
+    private async Task OnBanNotification(BanNotificationData payload)
+    {
+
+        if ((await _entryManager.ServerEntity).Id == payload.ServerId)
+            return;
+
+        if (!_playerManager.TryGetSessionById(new NetUserId(payload.PlayerId), out var player))
+            return;
+
+        var reason = _loc.GetString("ban-kick-reason");
+        _netManager.DisconnectChannel(player.Channel, reason);
+        _logger?.Info($"Kicked player {player.Name} ({player.UserId}) for {reason} through ban notification");
+    }
+
+    public void Dispose()
+    {
+        _tokenSource.Cancel();
+        if (_connection == null)
+            return;
+
+        _connection.Notification -= OnNotification;
+        _connection.Dispose();
+    }
+
+    private sealed class BanNotificationData
+    {
+        [JsonRequired, JsonPropertyName("player_id")]
+        public Guid PlayerId { get; init; }
```

Comment:
This needs to be a ban ID or something, so purely IP-bans also get covered.

Reply:
If you do, make sure servers have some form of batching so bulk data imports don't cause the servers to be really inefficient.

Reply:
I don't know if I want to put in the effort of doing that if even the original logic to kick people on the server the ban was made on doesn't kick them if it was an IP ban:

```cs
  // If we're not banning a player we don't care about disconnecting people
  if (target == null)
    return;
```

Reply:
Fair enough then